include/signalrclient/scheduler.h

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#pragma once
+
+#include <exception>
+#include <functional>
+#include <string>
+#include <chrono>
+
+namespace signalr
+{
+    typedef std::function<void()> signalr_base_cb;
+    typedef std::function<void(std::exception_ptr)> signalr_cb;
+    typedef std::function<void(std::string, std::exception_ptr)> signalr_message_cb;
+
+    struct scheduler
+    {
+        virtual void schedule(const signalr_cb& cb, std::chrono::milliseconds delay = std::chrono::milliseconds::zero())
+        {
+            schedule(cb, nullptr, delay);
```

Comment:
This should call the pure virtual function.

I'm still not convinced that we need anything other than the pure virtual function. This is an API that is designed to be implemented, not consumed, so why make it complicated? There might be some small perf benefit to not needing to allocate a closure, but I feel that YAGNI. And if you do need it, it can trivially be added back in a non-breaking way.